miden-lib/tests/common/mod.rs

```diff
@@ -0,0 +1,75 @@
+pub use crypto::{
+    hash::rpo::{Rpo256 as Hasher, RpoDigest as Digest},
+    FieldElement, StarkField,
+};
+pub use miden_lib::{memory, MidenLib};
+pub use miden_objects::{
+    assets::{Asset, FungibleAsset},
+    notes::{Note, NoteVault},
+    transaction::{ExecutedTransaction, ProvenTransaction, TransactionInputs},
+    Account, AccountId,
+};
+pub use processor::{
+    math::Felt, AdviceInputs, AdviceProvider, ExecutionError, MemAdviceProvider, Process,
+    StackInputs, Word,
+};
+use std::{env, fs::File, io::Read, path::Path};
+use stdlib::StdLibrary;
+
+pub mod data;
+pub mod procedures;
+
+pub const TX_KERNEL_DIR: &str = "sat";
+
+// TEST BRACE
+// ================================================================================================
+
+/// Loads the specified file and append `code` into its end.
+pub fn load_file_with_code(imports: &str, code: &str, dir: &str, file: &str) -> String {
+    let assembly_file = Path::new(env!("CARGO_MANIFEST_DIR")).join("asm").join(dir).join(file);
+
+    let mut module = String::new();
+    File::open(assembly_file).unwrap().read_to_string(&mut module).unwrap();
+    let complete_code = format!("{imports}{module}{code}");
+
+    // This hack is going around issue #686 on miden-vm
+    complete_code.replace("export", "proc")
+}
+
+/// Inject `code` along side the specified file and run it
+pub fn run_within_tx_kernel<A>(
```

Comment:
This is really nice - but I wonder if in the future we should try to generalize this functionality and put it into the test utils crate which is being worked on in https://github.com/0xPolygonMiden/miden-vm/pull/797.

Reply:
Yeah I agree